src/Test/TestCommanderBrenner.java

```diff
@@ -0,0 +1,164 @@
+
+    game.endGame();
+    return testPassed;
+  }
+
+  /** Utility method that checks whether there are any valid actions of this type for this unit in place */
+  public boolean canAct(Unit unitToTry, UnitActionFactory actionType)
+  {
+    return null != actionType.getPossibleActions(testMap, GamePath.stayPut(unitToTry), unitToTry);
+  }
+  private boolean testCOU()
+  {
+    setupTest();
+
+    Unit bopterOnFactory = addUnit(testMap, Brenner, UnitModel.AIR_TO_SURFACE, 1, 7);
+    Unit bopterOnHQ      = addUnit(testMap, Brenner, UnitModel.AIR_TO_SURFACE, 1, 8);
+    Unit infOnFactory    = addUnit(testMap, Brenner, UnitModel.LAND,           2, 7);
+    Unit infOnGrass      = addUnit(testMap, Brenner, UnitModel.LAND,           2, 8);
+    Unit infOnUnowned    = addUnit(testMap, Brenner, UnitModel.LAND,           7, 8);
+
+    Brenner.initTurn(testMap);
+    Brenner.army.money = 9001;
+    boolean testPassed = true;
+    DeployableCommander.DeployCOUFactory couer = Brenner.deployAction;
+
+    testPassed &= validate(Brenner.COUs.size() == 0, "    Brenner has a COU before doing anything");
+    testPassed &= validate(!canAct(bopterOnFactory, couer), "    Brenner can set his COU on an air unit on a factory.");
+    testPassed &= validate( canAct(bopterOnHQ     , couer), "    Brenner can't set his COU on an HQ.");
+    testPassed &= validate( canAct(infOnFactory   , couer), "    Brenner can't set his COU on a factory he owns.");
+    testPassed &= validate(!canAct(infOnGrass     , couer), "    Brenner can set his COU on grass he doesn't own.");
+    testPassed &= validate(!canAct(infOnUnowned   , couer), "    Brenner can set his COU on a factory he doesn't own.");
+
+    performGameAction(new DeployableCommander.DeployCOUAction(Brenner.deployAction, infOnFactory), game);
+    testPassed &= validate(!infOnFactory.isTurnOver, "    COUing cost the COU's turn.");
+    testPassed &= validate(Brenner.COUs.contains(infOnFactory), "    COUing didn't COU the COU.");
+    testPassed &= validate(9001 - Brenner.army.money == infOnFactory.getCost()/2, "    COUing didn't cost the expected amount.");
+    testPassed &= validate(!canAct(bopterOnHQ     , couer), "    Brenner can double-COU.");
+    testPassed &= validate(!canAct(infOnFactory   , couer), "    Brenner can COU his COU.");
+
+    return testPassed;
+  }
+
+  private boolean testAttackCharge()
+  {
+    setupTest();
+
+    // Add our combatants
+    Unit mech = addUnit(testMap, Brenner, UnitModel.MECH, 1, 1);
+    Unit COU  = addUnit(testMap, Brenner, UnitModel.TRANSPORT, 4, 2);
+    Unit aa   = addUnit(testMap, Brenner, UnitModel.SURFACE_TO_AIR, 2, 2);
+
+    // And our victim
+    Unit inf = addUnit(testMap, Patch, UnitModel.TROOP, 1, 2);
+
+    Brenner.initTurn(testMap);
+    Brenner.army.money = 9001;
+    boolean testPassed = true;
+    
+    performGameAction(new DeployableCommander.DeployCOUAction(Brenner.deployAction, COU), game);
```

Comment:
The transport/COU is at (4, 2) which is a grass tile. Shouldn't this action fail? Or does this just skip the "can I?" step and take Shia's advice?

Reply:
Yeah, I've been depending a lot lately on validation at the `UnitActionFactory` level, since that is what determines what goes up on the GUI (i.e. an invalid action that's constructed and executed will simply fail with no explanation, while an action that doesn't get constructed just won't show up)